src/public/admin.js

```diff
@@ -0,0 +1,50 @@
+document.addEventListener('DOMContentLoaded', function () {
+  const updatePriceBtn = document.getElementById('updatePriceBtn');
+  const newPriceInput = document.getElementById('newPrice');
+  const priceEditForm = document.getElementById('priceEditForm');
+  const successMessage = document.getElementById('successMessage');
+  const errorMessage = document.getElementById('errorMessage');
+
+  updatePriceBtn.addEventListener('click', async () => {
+    // Disable the price update button to prevent multiple clicks
+    updatePriceBtn.disabled = true;
+
+    // Clear success and error messages
+    successMessage.textContent = '';
+    errorMessage.textContent = '';
+
+    // Get the selected albums and the new price
+    const selectedAlbums = Array.from(
+      document.querySelectorAll('input[type="checkbox"]:checked')
+    ).map((checkbox) => checkbox.value);
+
+    const newPrice = newPriceInput.value;
+
+    // Send a request to the server to update prices
+    try {
+      const response = await fetch('/admin/api/albums', {
+        method: 'PATCH',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          albumIds: selectedAlbums,
+          newPrice: newPrice,
+        }),
```

Comment:
Currently, you're having the `updatePriceOfAlbums` albums from the album controller, handle this request. But it's expecting the request body shape to be different than what you're sending here.

<img width="666" alt="Screenshot 2023-08-26 at 10 30 37 PM" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/9094098/b0ce7f81-689f-4a26-bcce-8903a8a1e123">

<img width="453" alt="Screenshot 2023-08-26 at 10 30 40 PM" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/9094098/261ed390-5fbc-4057-8d9c-7f3637f9fa20">

You could either:
- have some code in the admin controller that transforms the req.body before calling `updatePriceOfAlbums(req, res)`
- write code in admin controller that just completely handles this request payload on its own
- update the code here in `admin.js` to send the request in the shape expected by `updatePriceOfAlbums` (if that endpoint is not yet documented, I believe Natalia implemented it so you can double check with her)

Reply:
fixed. Thank you for your help❤️